ts/sagas/wallet.ts

```diff
@@ -24,10 +90,203 @@ function* fetchTransactions(
+      codiceContestoPagamento: "6793ad707f9b11e888482902221575ae" as CodiceContestoPagamento,
+      ibanAccredito: "IT17X0605502100000001234567" as Iban,
+      causaleVersamento: "IMU 2018",
+      enteBeneficiario: {
+        identificativoUnivocoBeneficiario: "123",
+        denominazioneBeneficiario: "Comune di Canicattì",
+        codiceUnitOperBeneficiario: "01",
+        denomUnitOperBeneficiario: "CDC",
+        indirizzoBeneficiario: "Via Roma",
+        civicoBeneficiario: "23",
+        capBeneficiario: "92010",
+        localitaBeneficiario: "Canicattì",
+        provinciaBeneficiario: "Agrigento",
+        nazioneBeneficiario: "IT"
+      } as EnteBeneficiario
+    };
+    yield put(
+      paymentTransactionSummaryFromRptId(rptId, initialAmount, verificaResponse)
+    );
+  } else {
+    yield put(paymentTransactionSummaryFromBanner());
+  }
+  // also, show summary screen
+  yield put(navigateTo(ROUTES.PAYMENT_TRANSACTION_SUMMARY));
+}
+
+function* continueWithPaymentMethodsHandler(
+  _: PaymentRequestContinueWithPaymentMethods
+) {
+  // find out whether a payment method has already
+  // been defined as favorite. If so, use it and
+  // ask the user to confirm it
+  // Otherwise, show a list of payment methods available
+  // TODO: if no payment method is available (or if the
+  // user chooses to do so), allow adding a new one.
+  const favoriteCard: Option<number> = yield select(getFavoriteWalletId);
+  if (favoriteCard.isSome()) {
+    // show card
+    yield put(paymentRequestConfirmPaymentMethod(favoriteCard.value));
+  } else {
+    // show list
+    yield put(paymentRequestPickPaymentMethod());
+  }
+}
+
+function* confirmPaymentMethodHandler(
+  action: PaymentRequestConfirmPaymentMethod
+) {
+  const walletId = action.payload;
+  yield put(paymentConfirmPaymentMethod(walletId));
+  yield put(navigateTo(ROUTES.PAYMENT_CONFIRM_PAYMENT_METHOD));
+}
+
+function* pickPaymentMethodHandler(_: PaymentRequestPickPaymentMethod) {
+  // show screen with list of payment methods available
+  yield put(paymentPickPaymentMethod());
+  yield put(navigateTo(ROUTES.PAYMENT_PICK_PAYMENT_METHOD));
+}
+
+function* completionHandler(_: PaymentRequestCompletion) {
+  // -> it should proceed with the required operations
+  // and terminate with the "new payment" screen
+
+  // do payment stuff (-> pagoPA REST)
+  // retrieve transaction and store it
+  // mocked data here
+  // tslint:disable-next-line saga-yield-return-type
+  const wallet: Wallet = yield select(selectedPaymentMethodSelector);
+  // tslint:disable-next-line saga-yield-return-type
+  const recipient = (yield select(getPaymentRecipient)).getOrElse(
+    UNKNOWN_RECIPIENT
+  );
+  // tslint:disable-next-line saga-yield-return-type
+  const paymentReason = (yield select(getPaymentReason)).getOrElse(
+    UNKNOWN_PAYMENT_REASON
+  );
+  const fee = feeExtractor(wallet);
+
+  const transaction: WalletTransaction = {
+    id: Math.floor(Math.random() * 1000),
```

Comment:
whats is going on here?

Reply:
This data is currently mocked and needs to be fetched from the pagoPA API (this is done in a separate PR that depends on this one) -- the id is being randomly generated to (reasonably) avoid overlaps. In the actual integration, this transaction will be sent by pagoPA, with the id and everything